testkit-backend/backend.go

```diff
@@ -1440,8 +1446,11 @@ func serializeRecord(record *neo4j.Record) map[string]any {
 	return data
 }
 
-func serializeNotifications(slice []neo4j.Notification) []map[string]any {
+func serializeNotifications(slice []neo4j.Notification, version db.ProtocolVersion) []map[string]any {
 	if slice == nil {
+		if version.Minor >= 5 {
+			return []map[string]any{}
+		}
```

Comment:
I think we've had a short chat about it, but I can't remember: could you please explain to me why this special casing for some bolt version is required?

Also, I'm almost certain that this should not only depend on the protocol's minor version, but also or only on the major version.

Reply:
The conversation was pretty much:

> Before 5.5, when the server doesn't send notifications, our drivers behave differently. Some (I think most) return an empty list and they skip this test. Starting with 5.5, the server never sends notifications, but the driver is supposed to polyfill notifications from statuses. And if there are no notifications in the statuses, the driver should return an empty list. TL;DR: unspecified behavior before 5.5.

What was happening was that the Go driver returned nil when serializing notifications, not an empty slice. Now that we expect an empty slice after bolt 5.5, the Testkit tests also expect an empty slice. So this was to prevent older tests from breaking. You're right about the major version though, I've updated the check to include major 4 too.